File: Partie-Backend/src/main/java/com/basketapp/basket_app/models/Produit.java
package com.basketapp.basket_app.models;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;

import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

import com.sun.istack.NotNull;
import org.springframework.lang.Nullable;

import javax.persistence.GenerationType;
import javax.persistence.GeneratedValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Produit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idProduit;
    private String nomProduit;
    private Float prix ;
    private String description;
    private String couleur;
    private String taille;
    private String genre; 
    private Boolean favoris=false; 
    @Lob
    String image;

    @NotNull
    private Float Moy;

    @ManyToOne
    private Categorie categorie;

    
    @ManyToOne
    private Marque marque;

    @JsonIgnore
    @OneToMany(mappedBy = "produit",cascade = CascadeType.ALL)
    private Set<ProduitPanier> produitPaniers = new HashSet<>();

    @JsonIgnore
    @OneToMany(mappedBy = "produit",cascade = CascadeType.ALL)
     private Set<Rating> rating = new HashSet<>();

}

File: Partie-Mobile/lib/models/produit.dart
class Produit {
  int? idProduit;
  String? nomProduit;
  double? prix;
  String? description;
  String? couleur;
  String? taille;
  String? genre;
  bool? favoris;
  String? image;
  Categorie? categorie;
  Marque? marque;
  double? moy;

  Produit(
      {this.idProduit,
      this.nomProduit,
      this.prix,
      this.description,
      this.couleur,
      this.taille,
      this.genre,
      this.favoris,
      this.image,
      this.categorie,
      this.marque,
      this.moy});

  Produit.fromJson(Map<String, dynamic> json) {
    idProduit = json['idProduit'];
    nomProduit = json['nomProduit'];
    prix = json['prix'];
    description = json['description'];
    couleur = json['couleur'];
    taille = json['taille'];
    genre = json['genre'];
    favoris = json['favoris'];
    image = json['image'];
    categorie = json['categorie'] != null
        ? new Categorie.fromJson(json['categorie'])
        : null;
    marque =
        json['marque'] != null ? new Marque.fromJson(json['marque']) : null;
    moy = json['moy'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idProduit'] = this.idProduit;
    data['nomProduit'] = this.nomProduit;
    data['prix'] = this.prix;
    data['description'] = this.description;
    data['couleur'] = this.couleur;
    data['taille'] = this.taille;
    data['genre'] = this.genre;
    data['favoris'] = this.favoris;
    data['image'] = this.image;
    if (this.categorie != null) {
      data['categorie'] = this.categorie!.toJson();
    }
    if (this.marque != null) {
      data['marque'] = this.marque!.toJson();
    }
    data['moy'] = this.moy;
    return data;
  }
}

class Categorie {
  int? idCategorie;
  String? label;

  Categorie({this.idCategorie, this.label});

  Categorie.fromJson(Map<String, dynamic> json) {
    idCategorie = json['idCategorie'];
    label = json['label'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idCategorie'] = this.idCategorie;
    data['label'] = this.label;
    return data;
  }
}

class Marque {
  int? idMarque;
  String? nom;

  Marque({this.idMarque, this.nom});

  Marque.fromJson(Map<String, dynamic> json) {
    idMarque = json['idMarque'];
    nom = json['nom'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idMarque'] = this.idMarque;
    data['nom'] = this.nom;
    return data;
  }
}